src/posit/connect/oauth/integrations.py

```diff
@@ -1,56 +1,82 @@
+from .._utils import assert_guid
+from ..context import Context
+from ._types_context_integration import IntegrationContext
 from .associations import IntegrationAssociations
 
 
-class Integration(Resource):
+class Integration(ActiveDict[IntegrationContext]):
     """OAuth integration resource."""
 
+    def __init__(self, ctx: Context, /, *, guid: str, **kwargs):
+        guid = assert_guid(guid)
+
+        integration_ctx = IntegrationContext(ctx, integration_guid=guid)
+        path = f"v1/oauth/integrations/{guid}"
+        get_data = len(kwargs) == 0  # `guid` is required
+        super().__init__(integration_ctx, path, get_data, guid=guid, **kwargs)
+
     @property
     def associations(self) -> IntegrationAssociations:
-        return IntegrationAssociations(self.params, integration_guid=self["guid"])
+        return IntegrationAssociations(
+            self._ctx,
+        )
 
     def delete(self) -> None:
         """Delete the OAuth integration."""
         path = f"v1/oauth/integrations/{self['guid']}"
-        url = self.params.url + path
-        self.params.session.delete(url)
+        url = self._ctx.url + path
+        self._ctx.session.delete(url)
+
+    class _AttrsUpdate(TypedDict, total=False):
+        name: str
+        description: str
+        config: dict
 
-    @overload
     def update(
         self,
-        *args,
-        name: str = ...,
-        description: str = ...,
-        config: dict = ...,
-        **kwargs,
-    ) -> None:
+        **kwargs: Unpack[_AttrsUpdate],
+    ) -> "Integration":
         """Update the OAuth integration.
 
         Parameters
         ----------
         name: str, optional
+            A descriptive name to identify each OAuth integration.
         description: str, optional
+            A brief text to describe each OAuth integration.
         config: dict, optional
+            The OAuth integration configuration based on the template. See List OAuth templates for
+            more information on available fields for each template. The configuration combines
+            elements from both options and fields from a given template.
         """
+        result = self._patch_api(json=kwargs)
+        assert result is not None, "Integration update failed"
+        assert "guid" in result, "Integration update failed. No guid returned."
+        return Integration(self._ctx, **result)
 
-    @overload
-    def update(self, *args, **kwargs) -> None:
-        """Update the OAuth integration."""
 
-    def update(self, *args, **kwargs) -> None:
-        """Update the OAuth integration."""
-        body = dict(*args, **kwargs)
-        url = self.params.url + f"v1/oauth/integrations/{self['guid']}"
-        response = self.params.session.patch(url, json=body)
-        super().update(**response.json())
+# TODO-barret-future-q; Should this auto retrieve? If so, it should inherit from ActiveSequence
+class Integrations(ApiCallMixin, ContextP[Context]):
```

Comment:
I'm really enjoying the clarity of when data is retrieved from `.find()` / `.get()` / `.all()` methods. And it does take the ambiguity out of _when_ data is being retrieved. 

I'd be up for dropping the "download all list-like data on initialization" in favor of the retrieval methods (e.g. `.find()` etc.) . This would also mean that the classes wouldn't be `list-like` would require the user to retrieve before calling any list like methods (ex: `len(jobs.all())` and not `len(jobs)` which can change over time.